HowTo notebooks

mpmoine opened the pull request:
In this PR only changes in `notebooks/` directory with  8 new notebooks:

- HowTo-00_Install_the_DR_Software.ipynb
- HowTo-01_Import_and_Load_the_DR.ipynb
- HowTo-02_Discover_What_is_in_DR.ipynb
- HowTo-03a_Find_Experiments_and_Variables_for_given_Opportunities.ipynb
- HowTo-03b_Find_Experiments_and_Variables_for_given_Opportunities.ipynb
- HowTo-04a_View_Attributes_of_Experiments_and_Variables.ipynb
- HowTo-04b_View_Attributes_of_Variables.ipynb
- HowTo-05_Apply_Various_Search_Critria.ipynb

A few comments:
- HowTo-00: is a remake of the main README with a lot of in-prompt commands ->  not so userful I guess
- HowTo-01: more or less a remake of what is in the content/README
- HowTo-02: for retrieving full list of categories in the DR content (global discovery purpose)
- HowTo-03a & 3b: 2 alternatives for searching vars and expes requested by a given opp
- HowTo-04a & 4b: 2 alternatives for  for viewing expe and var attributes
- HowTo-05: serial execution of ‘find’ and ‘get’ methods appying. to DR object

To come:
- HowTo-06_Compare_Variable_Attributes_between_2_DR_versions.ipynb -> a notebook version of  compare_variable_metadata.py
- HowTo-07_Export_Results_in_Smart_Formats.ipynb 
- and may be one that uses HowTo-05 in a pipe/chain mode...
- or other any ideas... 

For sure these notebooks are very simple/naives but I guess they intent to be pedagogic for peaple droping for the first time in the software.

mpmoine committed:
MPM's first drafting jupyter-notebooks

mpmoine committed:
Merge branch 'main' into mpm_sandbox

mpmoine committed:
Merge branch 'main' into mpm_sandbox

mpmoine committed:
Merge remote-tracking branch 'origin_dreq/main' into howto_notebooks
update

mpmoine committed:
Merge remote-tracking branch 'origin_dreq/main' into howto_notebooks

mpmoine committed:
Merge remote-tracking branch 'origin_dreq/main' into howto_notebooks

mpmoine committed:
first HowTo notebooks (00 to 05)

mpmoine commented:
still to do:
- make some corrections (some pbs remains... or things I do not undertand correctly with DR version selection...)
- say in the README to install an ipykernel in the venv to be able to run the notebooks
- commit an empty `notebooks/outdir`

JamesAnstey commented:
These look great, thanks @mpmoine! 

There are be some inconsistencies with my recent changes in https://github.com/CMIP-Data-Request/CMIP7_DReq_Software/pull/101 where `dq.` functions are used although I'm not sure it matters. I can test.

sol1105 commented:
For a quick look:
https://nbviewer.org/github/mpmoine/CMIP7_DReq_Software/tree/howto_notebooks/notebooks/

We could also add a NBViewer link/button for the main branch notebooks to our Readme?

JamesAnstey commented on notebooks/HowTo-00_Install_the_DR_Software.ipynb:
```diff
@@ -0,0 +1,259 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Option-2 : Developper Installation mode\n",
```
Do we need developer install instructions in a notebook? Might be simpler to leave out, if notebooks are intended as tutorial for general users.

mpmoine replied:
You're right @JamesAnstey , I will remove  install instructions for the develop mode 👍

JamesAnstey commented on notebooks/HowTo-01_Import_and_Load_the_DR.ipynb:
```diff
@@ -0,0 +1,373 @@
+   "source": [
+    "# content = dc.load(version='v1.0')\n",
+    "use_dreq_version = 'latest_stable'\n",
+    "dq.DREQ_VERSION = use_dreq_version\n",
```
All `dq.DREQ_VERSION` lines should be deleted after merging in latest `main`, this is no longer used. (Also in subsequent notebooks.)

mpmoine replied:
good point, thanks for the notification @JamesAnstey 👍

JamesAnstey commented on notebooks/HowTo-01_Import_and_Load_the_DR.ipynb:
```diff
@@ -0,0 +1,373 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dq.version_base_name()"
```
`dq.version_base_name()` has been removed (it was getting the top-level dict key for a base, and was intended as an internal function anyways). So should delete here, and in any other notebooks having it.

mpmoine replied:
I have always a train late! 🫤 . Ok to remove that also.

JamesAnstey commented on notebooks/HowTo-03b_Find_Experiments_and_Variables_for_given_Opportunities.ipynb:
```diff
@@ -0,0 +1,128 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "database = dq.create_dreq_tables_for_request(content)"
```
This now has a mandatory version argument: 
```
create_dreq_tables_for_request(content, dreq_version)
```
because of dq.DREQ_VERSION being removed.

JamesAnstey commented on notebooks/HowTo-04b_View_Attributes_of_Variables.ipynb:
```diff
@@ -0,0 +1,161 @@
+   "outputs": [],
+   "source": [
+    "used_cmor_tables = ['Amon', 'Omon']\n",
+    "metadata1 = dq.get_variables_metadata(database, cmor_tables=used_cmor_tables, use_dreq_version=used_version)"
```
Also now has a mandatory version argument:
```
def get_variables_metadata(content, dreq_version,
                           compound_names=None, cmor_tables=None, cmor_variables=None):
```

JamesAnstey commented on notebooks/HowTo-04b_View_Attributes_of_Variables.ipynb:
```diff
@@ -0,0 +1,161 @@
+   "source": [
+    "# write result in a json file\n",
+    "json_out1 = 'var_attrs_2tables.json'\n",
+    "dq.write_variables_metadata(metadata1, json_out1, api_version='dev', use_dreq_version=used_version, content_path=path_to_content)"
```
Continuing the theme of mandatory version argument :)
```
def write_variables_metadata(all_var_info, dreq_version, filepath,
                             api_version=None, content_path=None):
```

JamesAnstey reviewed (approved):
Thanks again @mpmoine for these notebooks, I think they'll be quite helpful for users

mpmoine committed:
Merge remote-tracking branch 'origin_dreq/main' into howto_notebooks

mpmoine committed:
adaptations after merging with lastest main
- remove dq.DREQ_VERSION, dq.version_base_name()
- add mandatory arg to create_dreq_tables_for_request
- add mandatory arg to get_variables_metadata
- add/rename madatory args to write_variable_metadata
- other updating/cleaning of notebooks

mpmoine commented:
All @JamesAnstey 's advices taken into account in my last commit 🍻

mpmoine committed:
show quick summary and write results in json files (prefixed with notebook number)
